### PR TITLE
[platform] addition of QPG7015M/GP712 RCP build

### PR DESCRIPTION
**Pull request by tima-q**
* Addition of a RCP build for QPG7015M and GP712 Qorvo platforms
* Addition required RCP API to glue code

### CODE REVIEW

File: examples/Makefile-qpg7015m [hunk 1]
```diff
@@ -71,7 +76,6 @@ COMMONCFLAGS                             := \
     -Wextra                                 \
     -Wall                                   \
     -Wno-psabi                              \
-    $(NULL)
```
Comment (jwhui) [inline examples/Makefile-qpg7015m:74]: Any reason for removing this?

File: examples/platforms/gp712/entropy.c [hunk 2]
```diff
@@ -41,22 +41,22 @@
 
 #include "utils/code_utils.h"
 
-#if __SANITIZE_ADDRESS__ != 0
+#if defined(__SANITIZE_ADDRESS__) && __SANITIZE_ADDRESS__ != 0
```
Comment (jwhui) [inline examples/platforms/gp712/entropy.c:44]: Is this change necessary? Also, do you use the address sanitizer on this platform? If not, we could just remove these defines.

File: examples/platforms/gp712/radio.c [hunk 3]
```diff
@@ -77,6 +77,11 @@ typedef struct otCachedSettings_s
 
 static otCachedSettings_t otCachedSettings;
 
+/* Upper layer relies on txpower could be set before receive, but MAC have per-channel config for it.
+   Store txpower until channel set in Receive(). */
+#define PENDING_TX_POWER_NONE (-1)
+int8_t pendingTxPower = PENDING_TX_POWER_NONE;
```
Comment (jwhui) [inline examples/platforms/gp712/radio.c:83]: Declare as static?

File: examples/platforms/qpg6095/radio.c [hunk 4]
```diff
@@ -77,6 +77,11 @@ typedef struct otCachedSettings_s
 
 static otCachedSettings_t otCachedSettings;
 
+/* Upper layer relies on txpower could be set before receive, but MAC have per-channel config for it.
+   Store txpower until channel set in Receive(). */
+#define PENDING_TX_POWER_NONE (-1)
+int8_t pendingTxPower = PENDING_TX_POWER_NONE;
```
Comment (jwhui) [inline examples/platforms/qpg6095/radio.c:83]: Declare as static?

File: examples/platforms/qpg7015m/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
+# OpenThread on Qorvo qpg7015m Example
+
+This directory contains example platform drivers for Qorvo qpg7015m on RPi.
+
+## Toolchain
+
+This example use the GNU GCC toolchain on the Raspberry Pi. To build on the Pi:
```
Comment (jwhui) [inline examples/platforms/qpg7015m/README.md:7]:
```suggestion
This example uses the GNU GCC toolchain on the Raspberry Pi. To build on the Pi:
```

**Comment by nikita-s-wrk** [comment]
@googlebot I fixed it.

**Comment by nikita-s-wrk** [comment]
@googlebot I consent.

**Comment by nikita-s-wrk** [comment]
@googlebot I fixed it.

**Comment by nikita-s-wrk** [comment]
@googlebot I fixed it.

**Comment by nikita-s-wrk** [comment]
@googlebot I signed it!

**Comment by nikita-s-wrk** [comment]
@googlebot I signed it!

**Comment by nikita-s-wrk** [comment]
@googlebot I signed it!

**Comment by nikita-s-wrk** [comment]
@googlebot I signed it!

**Comment by jwhui** [comment]
Superseded by #6110
